File: fsharp/rna-transcription/RNATranscription.fs
module RNATranscription

let transform char =
    match char with
    | 'G' -> 'C'
    | 'C' -> 'G'
    | 'T' -> 'A'
    | 'A' -> 'U'
    | _ -> char

let toRna (dna: string) = dna |> String.map transform

File: typescript/rna-transcription/rna-transcription.ts
class Transcriptor {
  toRna(dna: string): string {
    const transforms: { [key: string]: string} = {
      G: "C",
      C: "G",
      T: "A",
      A: "U"
    }

    const rna = dna.replace(/./g, (x) => transforms[x])

    if (rna.length !== dna.length) {
      throw new Error("Invalid input DNA.")
    }

    return rna
  }
}

export default Transcriptor
